src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -226,12 +227,25 @@ public void Shutdown()
 
             if (projectImportsCollector != null)
             {
+                projectImportsCollector.Close();
```

Comment:
Why does this have to be up here instead of where it was before?

Reply:
So that we first flush the remaining archive bits to disk in the temp folder and close the file, so that we can read it later.